rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -507,7 +507,7 @@ class WriterInfo : public ReaderListener
   {
     WriterProxyData proxyData;
     if (change->kind == ALIVE) {
-      CDRMessage_t tempMsg;
+      CDRMessage_t tempMsg(0);
```

Comment:
I'm OK with this fix for now (since CDRMessage_t is not in our code), but it feels fragile to me.  For instance, it looks like the WriterInfo class has the same leak.  It seems to me that we should probably add (or get eProsima to add) another constructor to CDRMessage_t so that we can specify all of the pieces we want it to be without accessing the underlying fields directly.  Otherwise, I can see this occurring again.